### PR TITLE
CBL-6308: Implement MmapEnabled property on DatabaseConfiguration

### PR DESCRIPTION
**Pull request by borrrden**
Also sneaking in a few other small things here:

- Support for tool based test spec verification (ImplementsTestSpecAttribute)
- Complete ArrayIndexTest now that c4index_getOptions returns unnestPath
- Update LiteCore to 3.2.1-8

### CODE REVIEW

**Review by pasin — DISMISSED** [review]

File: src/Couchbase.Lite.Shared/API/Database/DatabaseConfiguration.cs [hunk 1]
```diff
@@ -107,7 +124,11 @@ public EncryptionKey? EncryptionKey
         /// </summary>
         public DatabaseConfiguration()
         {
-
+#if !MACCATALYST
+            if(!RuntimeInformation.IsOSPlatform(OSPlatform.OSX)) {
+                _mmapEnabled = Constants.DefaultDatabaseMmapEnabled;
+            }
+#endif
```
[the remark this answers is not in the document]
Comment (pasin) [reply src/Couchbase.Lite.Shared/API/Database/DatabaseConfiguration.cs:131]: After all, this is minor but I am wondering why not including `MACCATYLYST`?

[the remark this answers is not in the document]
Comment (pasin) [reply src/Couchbase.Lite.Shared/API/Database/DatabaseConfiguration.cs:131]: After all, this is minor but I am wondering why not including MACCATYLYST?

Comment (pasin) [inline src/Couchbase.Lite.Shared/API/Database/DatabaseConfiguration.cs:131]: From the CBL Discussion meeting, we don't need to implementation any guardrails or special logic for macOS. LiteCore will decide what to do. After all, this is minor but I am wondering why not including MACCATYLYST?

Comment (pasin) [inline src/Couchbase.Lite.Shared/API/Database/DatabaseConfiguration.cs:131]: From the CBL Discussion meeting, we don't need to implementation any guardrails or special logic for macOS. LiteCore will decide what to do. After all, this is minor but I am wondering why not including MACCATYLYST?

Comment (pasin) [inline src/Couchbase.Lite.Shared/API/Database/DatabaseConfiguration.cs:131]: From the CBL Discussion meeting, we don't need to implementation any guardrails or special logic for macOS. LiteCore will decide what to do. I am wondering why not including MACCATYLYST?

**Review by pasin — CHANGES_REQUESTED** [review]

**Comment by pasin** [comment]
Please ignore dismiss stale review comment above.

File: src/Couchbase.Lite.Shared/API/Database/DatabaseConfiguration.cs [hunk 2]
```diff
@@ -76,18 +82,29 @@ public bool FullSync
         }
 
         /// <summary>
-        /// Experiment API. Enable version vector.
+        /// Enables or disables memory-mapped I/O. By default, memory-mapped
```
Comment (bmeike) [inline src/Couchbase.Lite.Shared/API/Database/DatabaseConfiguration.cs:85]: I worded this differently, based on Jens' comments:
" Advises Core to enable or disable memory-mapped Database files, if possible."

**Review by bmeike — APPROVED** [review]
LGTM.  One suggestion on the docs, which you can accept or not, as you desire.

File: src/Couchbase.Lite.Shared/API/Database/DatabaseConfiguration.cs (hunk 1 above)
Comment (pasin) [reply src/Couchbase.Lite.Shared/API/Database/DatabaseConfiguration.cs:131]: After all, if you feel strongly that you want to do this, I don't have the problem at all.

**Review by pasin — APPROVED** [review]

File: src/Couchbase.Lite.Shared/API/Database/DatabaseConfiguration.cs (hunk 2 above)
Comment (borrrden) [reply src/Couchbase.Lite.Shared/API/Database/DatabaseConfiguration.cs:85]: Hmmm maybe, but it's also marked as unsupported on the two OS that it doesn't function on, which will give a compiler warning anyway.

**Review by pasin — APPROVED** [review]

**Pull request merged**